py/DREAM/Output/SPIShardPositions.py

```diff
@@ -15,6 +17,70 @@ def __init__(self, name, data, grid, output, attr=list()):
         super().__init__(name=name, data=data, attr=attr, grid=grid, output=output)
         
 
+    def arrivalTime(self):
+        """
+        Estimate the time at which the pellet arrives to the plasma edge.
+        """
+        if 'eq' not in self.grid:
+            raise OutputException("Cannot plot poloidal trajectory when equilibrium data is not stored in output.")
+
+        R0 = self.grid.eq.R0
+        if np.isinf(R0): R0 = 1
+
+        ROverR0 = self.grid.eq.ROverR0_f*R0 - R0
+        Z = self.grid.eq.Z_f - self.grid.eq.Z0
+        ntheta = self.grid.eq.theta.size
+
+        vertices = [(ROverR0[i,-1], Z[i,-1]) for i in range(ntheta)]
+        p = path.Path(vertices)
+
+        xp = self.data[:,0::3,0]
+        yp = self.data[:,1::3,0]
+
+        # Check if the shards start within the plasma
+        if p.contains_point((xp[0,0], yp[0,0])):
+            return 0
+
+        # Find the pellet which is travelling the fastest in the x direction
+        imax = np.argmax(np.abs(xp[-1,:] - xp[0,:]))
```

Comment:
Do all pellets have the same speed at all times? Otherwise, why not use `xp[1,:]` instead of `xp[-1,:]`?